### PR TITLE
Removing unnecessary `device=device` in modeling_llama.py

### PR DESCRIPTION
**Pull request by Liyang90**
Removing unnecessary `device=device`

# What does this PR do?

Removing unnecessary `device=device` in the second argument to `torch.full`.

`torch.full` expects a scaler for the second argument: https://pytorch.org/docs/stable/generated/torch.full.html
So if a device tensor is passed to it, the tensor needs to be synced and sent to CPU first. On TPU, this blocks the tracing of the current iteration that should be overlapped with graph execution of the previous iteration.

Fixes # (issue)

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [x] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Review by amyeroberts — COMMENTED** [review]
Thanks for adding this fix @Liyang90! 

This logic is common across many models in the codebase. Could you also update the other `_make_causal_mask` functions?

File: src/transformers/models/llama/modeling_llama.py [hunk 1]
```diff
@@ -46,7 +46,7 @@ def _make_causal_mask(
     Make causal mask used for bi-directional self-attention.
     """
     bsz, tgt_len = input_ids_shape
-    mask = torch.full((tgt_len, tgt_len), torch.tensor(torch.finfo(dtype).min, device=device), device=device)
+    mask = torch.full((tgt_len, tgt_len), torch.tensor(torch.finfo(dtype).min), device=device)
```
[the remark this answers is not in the document]
Comment (Liyang90) [reply src/transformers/models/llama/modeling_llama.py:49]: > If `torch.full` accepts a scalar, could we get away with not wrapping it in `torch.tensor`?

Yes, that works as well.

Let me find all occurrence of this issue, and update them together.

[the remark this answers is not in the document]
Comment (Liyang90) [reply src/transformers/models/llama/modeling_llama.py]: Updated

**Review by amyeroberts — APPROVED** [review]
Thanks for contributing this improvement and adding it across the models!

**Pull request merged**